python/tank/authentication/session_cache.py

```diff
@@ -462,6 +472,31 @@ def get_current_host():
     return host
 
 
+def _get_recent_items(document, recent_field, current_field, type_name):
+    """
+    Extract the list of recent items from the document.
+
+    If the recent_field is not set, then we'll simply return the current_field's
+    value. The recent_field will be empty when upgrading from an older core
+    that didn't support the recent users/hosts list.
+
+    :param object document: Document to extract information from
+    :param recent_field: Field from which we need to retrieve
+    """
+    # Extract the list of recent items.
+    items = document[recent_field]
+
+    # Then check if the current is part of the list. It's not? This is because
+    # an older core updated the file, but didn't know about the recent list and
+    # didn't update it. This is possible because since day one the authentication.yml
+    # has been treated as document with certain fields set and when the document
+    # is rewritten it is rewritten as is, except for the bits that were updated.
+    if document[current_field] and document[current_field] not in items:
```

Comment:
Ensures  changes back and forth of core version keep the content of the login dialog coherent.

Reply:
Maybe use `.get` for all of these for extra safety?